trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogRegister.java

```diff
@@ -0,0 +1,234 @@
+    return String.format("DROP CATALOG %s", name);
+  }
+
+  void registerCatalog(String name, GravitinoCatalog catalog) {
+    try {
+      String catalogFileName = String.format("%s/%s.properties", catalogStoreDirectory, name);
+      File catalogFile = new File(catalogFileName);
+      if (catalogFile.exists()) {
+        String catalogContents = Files.readString(catalogFile.toPath());
+        if (!catalogContents.contains(GRAVITINO_DYNAMIC_CONNECTOR + "=true")) {
+          throw new TrinoException(
+              GRAVITINO_DUPLICATED_CATALOGS,
+              "Catalog already exists, the catalog is not created by gravitino");
+        } else {
+          throw new TrinoException(
+              GRAVITINO_CATALOG_ALREADY_EXISTS,
+              String.format(
+                  "Catalog %s in metalake %s already exists",
+                  catalog.getName(), catalog.getMetalake()));
+        }
+      }
+
+      if (checkCatalogExist(name)) {
+        throw new TrinoException(
+            GRAVITINO_DUPLICATED_CATALOGS, "Catalog already exists with unknown reason");
+      }
+      String createCatalogCommand = generateCreateCatalogCommand(name, catalog);
+      executeSql(createCatalogCommand);
+      LOG.info("Register catalog {} successfully: {}", name, createCatalogCommand);
+    } catch (Exception e) {
+      String message = String.format("Failed to register catalog %s", name);
+      LOG.error(message);
+      throw new TrinoException(GRAVITINO_RUNTIME_ERROR, message, e);
+    }
+  }
+
+  boolean checkCatalogExist(String name) {
+    String showCatalogCommand = String.format("SHOW CATALOGS like '%s'", name);
+    Exception failedException = null;
+    try {
+      int retries = EXECUTE_QUERY_MAX_RETRIES;
+      while (retries-- > 0) {
+        try (Statement statement = connection.createStatement()) {
+          // check the catalog is already created
+          statement.execute(showCatalogCommand);
+          ResultSet rs = statement.getResultSet();
+          while (rs.next()) {
+            String catalogName = rs.getString(1);
+            if (catalogName.equals(name) || catalogName.equals("\"" + name + "\"")) {
+              return true;
+            }
+          }
+          return false;
+        } catch (Exception e) {
+          failedException = e;
+          LOG.warn("Execute command failed: {}, ", showCatalogCommand, e);
+          Thread.sleep(EXECUTE_QUERY_BACKOFF_TIME_SECOND * 1000);
+        }
+      }
+      throw failedException;
+    } catch (Exception e) {
+      throw new TrinoException(GRAVITINO_RUNTIME_ERROR, "Failed to check catalog exist", e);
+    }
+  }
+
+  void executeSql(String sql) {
+    try {
+      int retries = EXECUTE_QUERY_MAX_RETRIES;
+      Exception failedException = null;
+      while (retries-- > 0) {
+        try (Statement statement = connection.createStatement()) {
+          // check the catalog is already created
+          statement.execute(sql);
+          return;
```

Comment:
Unnecessary return.

Reply:
If there is no return, the function will throw an exception.